tests/orm/test_querybuilder.py

```diff
@@ -852,6 +852,59 @@ def test_empty_filters(self):
         qb = orm.QueryBuilder().append(orm.Data, filters={'or': [{}, {}]})
         assert qb.count() == count
 
+    from aiida.orm import AbstractCode, Computer, InstalledCode, PortableCode, QueryBuilder
+
+    def test_abstract_code_filtering(self):
+        """Test that querying for AbstractCode correctly returns all code instances.
+
+        This tests the fix for issue #6687, where QueryBuilder couldn't find codes
+        when looking for AbstractCode due to a node_type mismatch.
+        """
+        # Set up test environment
+        computer = orm.Computer(
+            label='test_computer_abstract',
+            hostname='localhost',
+            transport_type='core.local',
+            scheduler_type='core.direct',
+        ).store()
+
+        # Create test codes of different types
+        installed_code = orm.InstalledCode(
+            label='test_installed_abstract',
+            computer=computer,
+            filepath_executable='/bin/bash',
+        ).store()
+
+        portable_code = orm.PortableCode(
+            label='test_portable_abstract',
+            filepath_executable='/bin/bash',
```

Comment:
I think this needs to be fixed, please look at existing tests at tests/orm/data/code/test_portable.py
```suggestion
            filepath_paths='/bin/bash',
```